app/(gcforms)/[locale]/(form administration)/form-builder/[id]/responses/[statusFilter]/components/ManageFormAccessDialog/actions.ts

```diff
@@ -0,0 +1,123 @@
+"use server";
+
+import { authCheckAndThrow } from "@lib/actions";
+import { prisma } from "@lib/integration/prismaConnector";
+import { TemplateUser } from "./types";
+import { AccessControlError } from "@lib/privileges";
+import {
+  InvalidDomainError,
+  MismatchedEmailDomainError,
+  TemplateNotFoundError,
+  UserAlreadyHasAccessError,
+} from "@lib/invitations/exceptions";
+import { getTemplateWithAssociatedUsers, removeAssignedUserFromTemplate } from "@lib/templates";
+import { serverTranslation } from "@i18n";
+import { logMessage } from "@lib/logger";
+import { inviteUserByEmail } from "@lib/invitations/inviteUserByEmail";
+import { cancelInvitation as cancelInvitationAction } from "@lib/invitations/cancelInvitation";
+
+export const sendInvitation = async (emails: string[], templateId: string, message: string) => {
+  const { ability } = await authCheckAndThrow();
+  const { t } = await serverTranslation("manage-form-access");
+
+  const errors: string[] = [];
+
+  const invites = emails.map(async (email) => {
+    try {
+      await inviteUserByEmail(ability, email, templateId, message);
+    } catch (e) {
+      if (e instanceof UserAlreadyHasAccessError) {
+        errors.push(t("userAlreadyHasAccess", { email }));
+      }
+      if (e instanceof MismatchedEmailDomainError) {
+        errors.push(t("emailDomainMismatch", { email }));
+      }
+      if (e instanceof InvalidDomainError) {
+        errors.push(t("invalidEmail", { email }));
+      }
+      if (e instanceof TemplateNotFoundError) {
+        errors.push(t("templateNotFound", { templateId }));
+        throw e; // stop processing other emails
+      }
+      if (e instanceof AccessControlError) {
+        errors.push(t("accessControlError"));
+        throw e; // stop processing other emails
+      }
+      logMessage.error("Invitation failed", e);
+      errors.push(t("invitationFailed", { email }));
+    }
+  });
+
+  try {
+    await Promise.all(invites);
```

Comment:
The `Promise.all` will reject on the first promise rejection.  Is the goal to ensure you have all the errors returned for all attempted invites or is it ok to only get a subset if the other promises don't reject in time to update the `errors` array?  If so you can use `Promise.allSettled` that will return an array once all the promises reject or resolve.

Reply:
Yeah, I think allSettled is more appropriate https://github.com/cds-snc/platform-forms-client/pull/4392/commits/cee7009aa4be5c4408f4fe0ee65a7d614c5927f2